model/src/lectures/lectureDoc.ts

```diff
@@ -31,16 +31,27 @@ namespace LectureDisplay {
         quizModel: Quizzes.Quiz;
     }
 
+    export interface Question {
+        text: string,
+        //userID: string,
+        anonym: boolean,
+        likes: number,
+        postedTime: Timestamp,
+    }
+
     export interface Lecture {
         pdfUri: string;
         nbOfPages: number;
         availableToStudents: boolean;
         audioTranscript: { [pageNumber: number]: string; };
+        questions: Question[];
     }
 
     export const Functions = FunctionFolder("lectures", {
         addAudioTranscript: FunctionOf<{ courseId: string, lectureId: string, pageNumber: number, transcription: string; }, {}, 'invalid_arg' | 'error_firebase' | 'successfully_added'>("addAudioTranscript"),
+        createQuestion: FunctionOf<{ courseId: string, lectureId: string, question: Question; }, { id: string; }, 'invalid_arg' | 'error_firebase' | 'empty_question'>("createQuestion"),
+        updateQuestion: FunctionOf<{ id: string, question: Question; }, { id: string; }, 'invalid_id' | 'error_firebase' | 'empty_question'>("updateQuestion"),
```

Comment:
Same as before, passing `question` "as it is" is insecure.

Reply:
Yes now I will have to modify this as well thanks